android/app/src/main/java/com/expensify/chat/bootsplash/BootSplashActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.expensify.chat.bootsplash;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.expensify.chat.MainActivity;
+
+public class BootSplashActivity extends AppCompatActivity {
+
+  protected void forwardIntentToMainActivity(Intent intent) {
+    Intent intentCopy = new Intent(intent);
+
+    intentCopy.setClass(this, MainActivity.class);
+    intentCopy.putExtras(intent);
+    intentCopy.setData(intent.getData());
+    intentCopy.setAction(intent.getAction());
```

Comment:
It looks like `BootSplashActivity` simply clones the Intent, then passes it to `MainActivity`. So I'm a bit confused as to why `MainActivity` couldn't simply be set to the default launch Activity in the Android manifest? Am I missing something here?

Reply:
React native doesn't support `android:background` on `MainActivity`, explaining this design choice. (cf issue https://github.com/facebook/react-native/issues/9072).

We cannot use `windowBackground` either because it will appear on keyboard opening (cf https://github.com/zoontek/react-native-bootsplash/issues/150 - even if it has been set to `null` once the splash screen has been hidden)

Reply:
Interesting, thanks for sharing!